jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java

```diff
@@ -23,15 +23,320 @@
+        // When
+        List<String[]> result2 = EnvUtil.splitOnLastColon(list2);
+        // Then
+        assertTrue(result2.isEmpty());
+    }
+
+    @Test
+    public void testRemoveEmptyEntrieWhenNotNull(){
+        //Given
+        List<String>  string1 = new ArrayList<>();
+        string1.add(" set ");
+        string1.add(" set2  ");
+        string1.add("");
+        //When
+        List<String>  result1 = EnvUtil.removeEmptyEntries(string1);
+        //Then
+        assertArrayEquals( new String[]{"set", "set2"} ,result1.toArray());
+    }
+
+    @Test
+    public void testRemoveEmptyEntriesWhenNull(){
+        //Given
+        List<String>  string2 = new ArrayList<>();
+        string2.add(null);
+        //When
+        List<String>  result2 = EnvUtil.removeEmptyEntries(string2);
+        //Then
+        assertTrue(result2.isEmpty());
+    }
+
+
+    @Test
+    public void testSplitAtCommasAndTrimWhenNotNull(){
+        //Given
+        Iterable<String>  strings1 = Collections.singleton("hello,world");
+        //When
+        List<String> result1 = EnvUtil.splitAtCommasAndTrim(strings1);
+        //Then
+        assertEquals(2,result1.size());
+        assertEquals("world", result1.get(1));
+        }
+
+    @Test
+    public void testSplitAtCommasAndTrimWhenNull(){
+        //Given
+        Iterable<String>  strings2 = Collections.singleton(null);
+        //When
+        List<String> result2 = EnvUtil.splitAtCommasAndTrim(strings2);
+        //Then
+        assertTrue(result2.isEmpty());
+    }
+
+    @Test
+    public void testExtractFromPropertiesAsList() {
+        //Given
+        String string = "key";
+        Properties properties = new Properties();
+        properties.put("key.name","value");
+        properties.put("key.value","valu");
+        properties.put("art","id");
+        properties.put("note","bool");
+        properties.put("key._combine","bool");
+        //When
+        List<String> result = EnvUtil.extractFromPropertiesAsList(string,properties);
+        //Then
+        assertEquals(2,result.size());
+        assertArrayEquals(new String[]{"valu", "value"}, result.toArray());
+    }
+
+    @Test
+    public void testExtractFromPropertiesAsMap(){
+        //Given
+        String prefix = "key";
+        Properties properties = new Properties();
+        properties.put("key.name","value");
+        properties.put("key.value","valu");
+        properties.put("art","id");
+        properties.put("note","bool");
+        properties.put("key._combine","bool");
+        //when
+        Map<String, String> result = EnvUtil.extractFromPropertiesAsMap(prefix,properties);
+        //Then
+        assertEquals(2 ,result.size());
+        assertEquals("value",result.get("name"));
+    }
+
+    @Test
+    public void testFormatDurationTill() throws InterruptedException {
+        long startTime = System.currentTimeMillis();
+        // Wait for 200 milliseconds
+        Thread.sleep(200L);
```

Comment:
This is very bad, try not to use this in tests ever. I don't even see the purpose of this sleep/wait in the test.

Reply:
I suggested him to use `Thread.sleep()` since the test is measuring time duration since a specified value

Reply:
Never use `Thread.sleep` in tests!

For this test, I don't even understand why we are using `system.currentTimeMillis();`, but if this is really necessary, then you can simply replace by:
```suggestion
long startTime = System.currentTimeMillis() - 200L;
```

Some quick sources with reasons why (although there are maaaany bad reasons to do so)
- http://www.kyleblaney.com/junit-best-practices#:~:text=Do%20not%20use%20Thread.&text=When%20a%20unit%20test%20uses,the%20production%20code%20is%20broken. 
- https://laurentkempe.com/2012/11/20/Using-ThreadSleep-in-unit-test-A-good-idea/

Reply:
ohk :neutral_face: .